tests/test_trainer.py

```diff
@@ -410,6 +410,74 @@ class CurrentTestModel(
         tutils.run_prediction(dataloader, trainer.model)
 
 
+def test_resume_from_checkpoint_epoch_restored(tmpdir):
+    """Verify resuming from checkpoint runs the right number of epochs"""
+    import types
+
+    tutils.reset_seed()
+
+    hparams = tutils.get_hparams()
+
+    def new_model():
+        # Create a model that tracks epochs and batches seen
+        model = LightningTestModel(hparams)
+        model.num_epochs_seen = 0
+        model.num_batches_seen = 0
+
+        def increment_epoch(self):
```

Comment:
the self is confusing...

Reply:
We're binding the function to a class method, so it has a self. Other than modifying the base `LightningTestModel`, I'm not sure how else to do it.